### PR TITLE
Feature: control python version upon install, set max version to gym

### PR DESCRIPTION
**Pull request by royerk**
## What

Given the pickiness of the entire setup in order to be consistent with kaggle's env:

* check python version upon running `python setup.py install`
* install a version of gym `< 0.20.0` to enforce compatibility with SB3

This should save some trouble to new users or when the repo is updated & a re-install is required.

## Test

* [x] install from scratch in an empty virtualenv, gym version is ok
* [x] install with an invalid python version to trigger error
* [x] run `train.py` successfully

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,12 @@
+import sys
 from setuptools import setup, find_packages
 
+if sys.version_info < (3,7):
+    sys.exit('Sorry, Python < 3.7 is not supported, upgrade to >=3.7, <3.8')
+
+if sys.version_info > (3,7):
+    sys.exit('Sorry, Python > 3.7 is not supported, downgrade to >=3.7, <3.8')
```
Comment (glmcdona) [inline setup.py:8]: Python 3.7 is just required if users want to use the package to submit agents to kaggle using our framework. Some users may be just using the python game environment for their own use-cases not using the RL agent framework we've added.

Maybe just add a warning message in this case? Maybe also add a warning to the train.py script that the models won't work in submissions unless 3.7 is used?

**Review by glmcdona — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (royerk) [reply setup.py:8]: Sounds good, gimme a moment :+1:

Comment (royerk) [reply setup.py]: Ok, this should work both on linux and windows.

**Comment by royerk** [comment]
I have made a mess in my git. Allow me to close this one and open a clean PR without conflicts. Really sorry about that.

**Pull request closed without merging**